web/components/notifications/notification-types.tsx

```diff
@@ -0,0 +1,899 @@
+      icon={
+        <AvatarNotificationIcon notification={notification} symbol={'✏️'} />
+      }
+      subtitle={subtitle}
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        <span>
+          {action}{' '}
+          {!isChildOfGroup && (
+            <PrimaryNotificationLink text={sourceContractTitle} />
+          )}
+          {isChildOfGroup && <>the question</>}
+        </span>
+      </>
+    </NotificationFrame>
+  )
+}
+
+function CommentNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, highlighted, isChildOfGroup } = props
+  const {
+    sourceUserName,
+    sourceUserUsername,
+    reason,
+    sourceText,
+    sourceContractTitle,
+  } = notification
+  const reasonText =
+    reason === 'reply_to_users_answer' || reason === 'reply_to_users_comment'
+      ? 'replied to you '
+      : `commented `
+  return (
+    <NotificationFrame
+      notification={notification}
+      isChildOfGroup={isChildOfGroup}
+      highlighted={highlighted}
+      icon={
+        <AvatarNotificationIcon notification={notification} symbol={'💬'} />
+      }
+      subtitle={truncateText(sourceText, 'md')}
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        {reasonText}
+        {!isChildOfGroup && (
+          <span>
+            on <PrimaryNotificationLink text={sourceContractTitle} />
+          </span>
+        )}
+      </>
+    </NotificationFrame>
+  )
+}
+
+function AnswerNotification(props: {
+  notification: Notification
+  highlighted: boolean
+  isChildOfGroup?: boolean
+}) {
+  const { notification, highlighted, isChildOfGroup } = props
+  const {
+    sourceUserName,
+    sourceUserUsername,
+    sourceText,
+    sourceContractTitle,
+  } = notification
+  return (
+    <NotificationFrame
+      notification={notification}
+      isChildOfGroup={isChildOfGroup}
+      highlighted={highlighted}
+      icon={
+        <AvatarNotificationIcon notification={notification} symbol={'🙋'} />
+      }
+      subtitle={truncateText(sourceText, 'lg')}
+      link={getSourceUrl(notification)}
+    >
+      <>
+        <UserLink
+          name={sourceUserName || ''}
+          username={sourceUserUsername || ''}
+          className={'relative flex-shrink-0 hover:text-indigo-500'}
+        />{' '}
+        answered{' '}
+        {!isChildOfGroup && (
+          <span>
+            on <PrimaryNotificationLink text={sourceContractTitle} />
+          </span>
+        )}
+      </>
+    </NotificationFrame>
+  )
+}
+
+function TaggedUserNotification(props: {
```

Comment:
comment w/ a tag should probably be a tag notification
<img width="432" alt="Screenshot 2022-11-29 at 3 39 39 PM" src="https://user-images.githubusercontent.com/23196210/204664017-23a6692f-52ef-43c8-90ed-ecb7ed85b126.png">